app/src/main/java/org/kiwix/kiwixmobile/zim_manager/local_file_transfer/TransferProgressFragment.java

```diff
@@ -0,0 +1,54 @@
+package org.kiwix.kiwixmobile.zim_manager.local_file_transfer;
+
+import android.os.Bundle;
+
+import androidx.fragment.app.Fragment;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import org.kiwix.kiwixmobile.R;
+
+import java.util.ArrayList;
+
+/**
+ * Part of the local file sharing module, this fragment is used to display the progress of the
+ * file transfer. It displays a list of files along with their current status (as defined in the
+ * {@link FileItem} class.
+ * */
+public class TransferProgressFragment extends Fragment {
+
+  private ArrayList<FileItem> fileItems;
+  private RecyclerView filesRecyclerView;
+  private FileListAdapter fileListAdapter;
+
+  public TransferProgressFragment() {
+    // Required empty public constructor
+  }
+
+  public TransferProgressFragment(ArrayList<FileItem> fileItems) {
+    this.fileItems = fileItems;
+  }
+
+  @Override
+  public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+    View view = inflater.inflate(R.layout.fragment_transfer_progress, container, false);
```

Comment:
butterknife.
Also as an aside before people started using butterknife it was common to override `onViewCreated` as it kept the creation and setup of your view seperated

Reply:
[Done](https://github.com/kiwix/kiwix-android/pull/1276/commits/28840898e008c346ac4b5b562eec5026cae3c28d#diff-a0f57270aa23e27cd4cd2d0cf244cb42R44)
Can butterknife binding be performed later, in `onViewCreated` instead?

Reply:
It is best to do it at the point closest to the view inflation but you could call it anywhere, just don't expect your bound views to have a value until you call bind, it is very much like field injection in this regard